### DISCUSSION TITLE
2021-10-18 Core Team Update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#114] [Core]
# Highlights:
Testnet sharding release delayed due to memory issues. Debugging in progress.

# Done:
@Longarithm  - PR that limits number of functions to 10k (#4954) (+ research around it - currently in mainnet the max contract has around 5k functions).
@EgorKulikov - massive sync test
@mzhangmzz - sharding upgrade landed on betanet, did loadtests on betanet.

# Progress:

@bowenwang1996  @mzhangmzz - working on enabling sharding in testnet (including some lower-priority issues around sharding)
@yordanmadzhunkov: fuzzing tests & validator selection: making tests deterministic by injecting proper mocks
@EgorKulikov - moving computation heavy parts into separate threads, learning more about the garbage collection. Fixing issues with too-long running tests.
@abacabadabacaba  - continued work on AVL tree based representation.(generating proofs for cases etc).
@mm-near - learning about the ecosystem, getting all the pieces to run together on localnet.
@Longarithm  - updating function call fees to reflect gas usage. Removing RuntimeConfig from GenesisConfig

**Comment by maxzaver** [comment]
@EgorKulikov , @mm-near what "massive sync test" is referring to?

**Reply by bowenwang1996** [reply]
This one https://github.com/near/nearcore/pull/5035
